packages/babel-plugin-transform-destructuring/src/index.js

```diff
@@ -44,13 +44,16 @@ export default declare((api, options) => {
     return false;
   }
 
-  const arrayUnpackVisitor = {
-    ReferencedIdentifier(path, state) {
-      if (state.bindings[path.node.name]) {
-        state.deopt = true;
-        path.stop();
-      }
-    },
+  // NOTE: This visitor is meant to be used via t.traverseFast
+  const arrayUnpackVisitor = (node, state, parent) => {
+    if (
+      t.isIdentifier(node) &&
+      t.isReferenced(node, parent) &&
+      state.bindings[node.name]
+    ) {
+      state.deopt = true;
+      return true;
```

Comment:
I'm not a huge fan of `return true` in traversal-like APIs because it's not clear if the behavior is `.skip()`-like or `.stop()`-like.

My preference would probably be for it to either:
* Explicitly return a STOP constant
* Skip the return value and throw an exception instead, e.g.
```
const STOP = {};
// ...
throw STOP;
```
and do
```
try {
  t.traverseFast(arr, arrayUnpackVisitor, state);
} catch (err) {
  if (err !== STOP) throw err;
}
```